File: server/models/forms/personal_forms/employment_details.go
package personal_forms

import "time"

type EmploymentDetail struct {
	EmploymentType     string  `form:"employmentType" json:"employment_type"`
	EmployerStatus     string  `form:"employerStatus" json:"employer_status"`
	Designation        string  `form:"designation" json:"designation"`
	ModeOfSalary       string  `form:"modeOfSalary" json:"mode_of_salary"`
	GrossMonthlyIncome string  `form:"grossMonthlyIncome" json:"gross_monthly_income"`
	NetMonthlyIncome   string  `form:"netMonthlyIncome" json:"net_monthly_income"`
	ProfileId 		string  `form:"profileId" json:"profile_id"`
	ApplicationId 		string  `form:"applicationId" json:"application_id"`
	Timestamp          time.Time `json:"timestamp"`
	FormId             string  `json:"form_id"`
}

File: aarthik_setu/lib/models/profiles/personal/forms/employment_details.dart
class EmploymentDetails {
  String employmentType;
  String employerStatus;
  String designation;
  String modeOfSalary;
  String grossMonthlyIncome;
  String netMonthlyIncome;
  String? id;
  String profileId;
  DateTime timestamp;
  String applicationId;

  EmploymentDetails({
    required this.employmentType,
    required this.employerStatus,
    required this.designation,
    required this.modeOfSalary,
    required this.grossMonthlyIncome,
    required this.netMonthlyIncome,
    required this.profileId,
    required this.timestamp,
    required this.applicationId,
    this.id,
  });

  Map<String, dynamic> toJson() {
    return {
      'employmentType': employmentType,
      'employerStatus': employerStatus,
      'designation': designation,
      'modeOfSalary': modeOfSalary,
      'grossMonthlyIncome': grossMonthlyIncome,
      'netMonthlyIncome': netMonthlyIncome,
      'profileId': profileId,
      'timestamp': timestamp,
      'applicationId': applicationId,
      'id': id,
    };
  }

  factory EmploymentDetails.fromJson(Map<String, dynamic> json) {
    return EmploymentDetails(
      employmentType: json['employmentType'],
      employerStatus: json['employerStatus'],
      designation: json['designation'],
      modeOfSalary: json['modeOfSalary'],
      grossMonthlyIncome: json['grossMonthlyIncome'],
      netMonthlyIncome: json['netMonthlyIncome'],
      profileId: json['profileId'],
      timestamp: json['timestamp'],
      applicationId: json['applicationId'],
      id: json['id'],
    );
  }
}
